calico_containers/pycalico/ipam.py

```diff
@@ -322,7 +322,7 @@ def _compare_and_swap_handle(self, handle):
                 _log.debug("Handle %s is not empty.", handle.handle_id)
                 try:
                     self.etcd_client.update(handle.update_result())
-                except ValueError:
+                except EtcdCompareFailed:
                     raise CASError(handle.handle_id)
         else:
             _log.debug("CAS Write new handle %s", handle.handle_id)
```

Comment:
No, I think they only trigger when the block has a db_result to compare against (update not write).